### ISSUE TITLE
CA1716: Namespace suppression in GlobalSuppressions.cs not working

### ISSUE THREAD

**Issue by FantasyTeddy** [dotnet/roslyn-analyzers#1666]
#### Analyzer package

Microsoft.CodeQuality.Analyzers, version [2.6.0](https://www.nuget.org/packages/Microsoft.CodeQuality.Analyzers/2.6.0)

#### Analyzer

IdentifiersShouldNotMatchKeywords

#### Behavior

Suppressing the message that a namespace which is named like an identifier does not work in the GlobalSuppressions.cs file.
The corresponding error/warning message still appears.

#### Example
`[assembly: System.Diagnostics.CodeAnalysis.SuppressMessage("Microsoft.Naming", "CA1716:IdentifiersShouldNotMatchKeywords", MessageId = "Error", Scope = "namespace", Target = "Some.Namespace.With.Keyword.Error")]`

**Comment by sharwell** [comment]
💭 It's possible this is a dotnet/roslyn bug, but I'm not completely sure about how the design applies here.

**Comment by mavasani** [comment]
The issues seems to be that the analyzer is reporting a diagnostic with no location, hence symbol based suppression does not work. I will fix the analyzer to report diagnostic with symbol location.

**Comment by savbace** [comment]
I still can't suppress rule for entire namespace using `GlobalSuppressions.cs`.

**Analyzer package**
Microsoft.CodeQuality.Analyzers, version **2.6.2**

**Rule**
CA1028:Enum Storage should be Int32

**Example**
```csharp
[assembly: System.Diagnostics.CodeAnalysis.SuppressMessage(
    "Design",
    "CA1028:Enum Storage should be Int32",
    Justification = "<Pending>",
    Scope = "namespace",
    Target = "Some.Namespace")]
```

**Comment by mavasani** [comment]
@savbace This is a dupe of https://github.com/dotnet/roslyn/issues/486. `namespace` suppression scope does not suppress diagnostics within all types/namespaces within a namespace. You will need to use the newly added `NamespaceAndDescendants` scope for this functionality. Note that this scope was added very recently, and will be first available with VS2019 Preview 2. See https://docs.microsoft.com/en-us/visualstudio/releases/2019/release-notes-preview for available previews.
